disttask/framework/dispatcher/register.go

```diff
@@ -15,14 +15,16 @@
 package dispatcher
 
 import (
+	"context"
+
 	"github.com/pingcap/tidb/disttask/framework/proto"
 	"github.com/pingcap/tidb/util/syncutil"
 )
 
 // TaskFlowHandle is used to control the process operations for each global task.
 type TaskFlowHandle interface {
-	ProcessNormalFlow(d Dispatch, gTask *proto.Task) (metas [][]byte, err error)
-	ProcessErrFlow(d Dispatch, gTask *proto.Task, receive string) (meta []byte, err error)
+	ProcessNormalFlow(ctx context.Context, h Handle, gTask *proto.Task) (metas [][]byte, err error)
```

Comment:
the comment says this is for each task, so why we need to specify `gTask` as argument?

Reply:
PTAL @zimulala , maybe let `RegisterTaskFlowHandle` register a factory function